### PR TITLE
[AArch64] Add support for Cortex-A725 and Cortex-X925

### PR DESCRIPTION
**Pull request by jthackray**
Cortex-A725 and Cortex-X925 are Armv9.2 AArch64 CPUs.

Technical Reference Manual for Cortex-A725:
   https://developer.arm.com/documentation/107652/latest

Technical Reference Manual for Cortex-X925:
   https://developer.arm.com/documentation/102807/latest

### CODE REVIEW

**Review by tommat01 — APPROVED** [review]
LGTM

File: llvm/lib/Target/AArch64/AArch64Processors.td [hunk 1]
```diff
@@ -723,6 +746,9 @@ def ProcessorFeatures {
                                  FeaturePerfMon, FeatureETE, FeatureTRBE,
                                  FeatureSPE, FeatureMTE, FeatureSVE2BitPerm,
                                  FeatureFP16FML, FeatureSPE_EEF];
+  list<SubtargetFeature> X925 = [HasV9_2aOps, FeatureMTE, FeatureFP16FML,
```
Comment (davemgreen) [inline llvm/lib/Target/AArch64/AArch64Processors.td:749]: Should this include FeatureSVE2BitPerm? It is included in the AEK_ list, and X4 the features.

Same for the A725 features.

Comment (jthackray) [reply llvm/lib/Target/AArch64/AArch64Processors.td:749]: Thanks, good spot. Fixed.

**Review by davemgreen — APPROVED** [review]
Thanks!

**Pull request merged**